src/openrct2/scripting/bindings/world/ScTileElement.cpp

```diff
@@ -351,6 +351,27 @@ namespace OpenRCT2::Scripting
         }
     }
 
+    DukValue ScTileElement::rideType_get() const
+    {
+        auto ctx = GetContext()->GetScriptEngine().GetContext();
+        auto el = _element->AsTrack();
+        if (el != nullptr)
+            duk_push_int(ctx, el->GetRideType());
+        else
+            duk_push_null(ctx);
+        return DukValue::take_from_stack(ctx);
+    }
+    void ScTileElement::rideType_set(uint16_t value)
+    {
+        ThrowIfGameStateNotMutable();
+        auto el = _element->AsTrack();
+        if (el != nullptr)
+        {
+            el->SetRideType(value);
```

Comment:
Should we perhaps be doing the validation of the value. Ensure its less than type count, ensure that allow arbitrary ride type cheat enabled.

Reply:
Then we should do the same in trackType_set, right?

Reply:
In general, there isn't much validation going on in ScTileElement.
IMO, whoever is fiddling with the data directly instead of using GameActions should know what he is doing.

Reply:
As changing the RideType to a value > the type count will cause an instant crash i think we have to at least do that.

Reply:
Actually there are tons of API calls that will cause an instant crash when done with the wrong parameters.
I don't see why this one is any different.

Reply:
Well then its good to start adding validation code. Scripts shouldn't be able to crash the game, if there are ways to do that currently it should be fixed.

Reply:
> As changing the RideType to a value > the type count will cause an instant crash i think we have to at least do that.

We should fix all the instances of this.

Reply:
Added `value < RIDE_TYPE_COUNT` check.